code/game/objects/items/trader.dm

```diff
@@ -1018,3 +1018,150 @@ var/global/list/alcatraz_stuff = list(
 					var/mob/living/living_mover = mover
 					living_mover.put_in_hands(I)
 				to_chat(mover,"<span class='good'>\The [src] dispenses a reward!</span>")
+
+
+//Mystery mob cubes//////////////
+
+/obj/item/weapon/storage/box/mysterycubes
+	name = "mystery cube box"
+	desc = "Dehydrated friends!"
+	icon = 'icons/obj/pbag.dmi'
+	icon_state = "pbag"	//Supposed to look kind of shitty, cubes aren't even wrapped
+	foldable = /obj/item/weapon/paper
+	can_only_hold = list("/obj/item/weapon/reagent_containers/food/snacks/monkeycube/mysterycube")
+
+/obj/item/weapon/storage/box/mysterycubes/New()
+	..()
+	var/friendAmount = 1
+	friendAmount = rand(1, 3)
+	for(var/i = 1 to friendAmount)
+		new /obj/item/weapon/reagent_containers/food/snacks/monkeycube/mysterycube(src)
+
+/obj/item/weapon/reagent_containers/food/snacks/monkeycube/mysterycube
+	name = "mystery cube"
+	desc = "A portable friend!"
+	var/list/potentialFriends = list()
```

Comment:
I'd do one of the following:
1. Make this a `static` var and only set it in `New()` if it hasn't already been set
2. Remove this var completely and just directly do `contained_mob = pick(existing_typesof(/mob/living/simple_animal) - (boss_mobs + blacklisted_mobs))`

Doesn't really matter which but the latter seems slightly easier